src/Common/src/Interop/Windows/kernel32/Interop.CreateNamedPipeClient.Uap.cs

```diff
@@ -31,8 +31,12 @@ internal partial class Kernel32
             Interop.Kernel32.CREATEFILE2_EXTENDED_PARAMETERS parameters;
             parameters.dwSize = (uint)Marshal.SizeOf<Interop.Kernel32.CREATEFILE2_EXTENDED_PARAMETERS>();
 
+            // The dwFlagsAndAttributes is carrying a combination of flags that are mapped to different fields of the extended
+            // parameters. The possible range of values for dwFileAttributes, dwSecurityQosFlags, and dwFileFlags cannot be fully
+            // covered coming from a single int but are enough for correction creation of the named pipe client. The SECURITY_VALID_SQOS_FLAGS
+            //  needs to be all available for proper impersonation.
             parameters.dwFileAttributes = (uint)dwFlagsAndAttributes & 0x0000FFFF;
-            parameters.dwSecurityQosFlags = (uint)dwFlagsAndAttributes & 0x000F0000;
+            parameters.dwSecurityQosFlags = (uint)dwFlagsAndAttributes & 0x001F0000; // SECURITY_VALID_SQOS_FLAGS
```

Comment:
why not put `private const uint SECURITY_VALID_SQOS_FLAGS = 0x001F0000;` in this file? it doesn't need to go everywhere. I don't see a good reason to have magic numbers, even with comments

Reply:
np

Reply:
but as general rule on the smallest possible scope, so it'll be a local constant since it is not used anywhere else. There is a chance that this will become the top hit if someone searches for `SECURITY_VALID_SQOS_FLAGS` ...